src/main/java/io/supertokens/webserver/WebserverAPI.java

```diff
@@ -378,6 +376,23 @@ protected StorageAndUserIdMapping enforcePublicTenantAndGetStorageAndUserIdMappi
         }
     }
 
+    protected StorageAndUserIdMapping getStorageAndUserIdMappingForAppSpecificApi(
+            HttpServletRequest req, String userId, UserIdType userIdType, boolean isCallFromAuthRecipeAPI)
+            throws StorageQueryException, TenantOrAppNotFoundException, UnknownUserIdException, ServletException {
+        AppIdentifier appIdentifier = getAppIdentifierWithoutVerifying(req);
+        Storage[] storages = StorageLayer.getStoragesForApp(main, appIdentifier);
+        try {
+            return StorageLayer.findStorageAndUserIdMappingForUser(
+                    appIdentifier, storages, userId, userIdType);
+        } catch (UnknownUserIdException e) {
+            if (isCallFromAuthRecipeAPI) {
+                throw e;
+            }
+
+            return new StorageAndUserIdMapping(getTenantStorage(req), null);
```

Comment:
no.. we need to get public tenant storage only! Not tenant from request.. The only diff between this function and the enforcePublicTenantAndGetStorageAndUserIdMappingForAppSpecificApi is that we do not enforce public tenant, but we still only operate on the public tenant.